app/routers/weight.py

```diff
@@ -0,0 +1,64 @@
+
+from fastapi import APIRouter, Depends, Request
+from sqlalchemy.orm import Session
+from starlette.responses import RedirectResponse
+
+from app.database.models import User
+from app.dependencies import get_db
+from app.dependencies import templates
+
+
+router = APIRouter(tags=["weight"],)
+
+
+@router.get("/weight")
+async def weight_form(
+        request: Request,
+        session: Session = Depends(get_db),
+        target: float = None,
+        current_weight: float = None,
+        ):
+    user = session.query(User).filter_by(id=1).first()
+    target = user.target_weight
+    if current_weight:
+        return RedirectResponse(url='/')
```

Comment:
you meant if not?

Reply:
No.
If we are using "not", it redirects at the page before moving to weight page.

Reply:
This seems to redirect to the home page is that what you meant?
Also, the logic isn't good  it seems you are asking if there is a current weight, if there is  return to the homepage.